File: python/剑指offer_10_1_斐波那契数列.py
class Solution:
    def fib(self, n: int) -> int:
        MOD = 10**9 + 7
        if n < 2:
            return n
        x, y, z = 0, 0, 1
        for i in range(2, n+1):
            x, y = y, z
            z = (x + y) % MOD
        return z

File: cpp/剑指offer_10_1_斐波那契数列.cpp
class Solution {
public:
    int fib(int n) {
        const int MOD = 1000000007;
        if (n < 2)
            return n;
        int x = 0, y = 0, z = 1;
        for (int i=2; i<n+1; i++) {
            x = y;
            y = z;
            z = (x + y) % MOD;
        }
        return z;
    }
};